lib/request.js

```diff
@@ -66,6 +66,8 @@ function doRequest (options, data) {
       // Even though executed by name, the first argument is still "node",
       // the second the script name. The third is the string we want.
       data = Buffer.from(process.argv[2] || '', encoding)
+      // There will be a trailing \n from the user hitting enter. Get rid of it.
+      data = data.replace(/\n$/, '')
```

Comment:
This code is obviously for TTY input, and it was located for non-TTY input.  As a result of being applied in the wrong place, the actual length of the test string, 30, was reduced in the test to 29.

Reply:
I've extracted this change in #29 
Could you please revert it in this pull request?